pkg/tsdb/prometheus/time_series_query.go

```diff
@@ -274,24 +272,32 @@ func calculateRateInterval(interval time.Duration, scrapeInterval string, interv
 	return rateInterval
 }
 
-func interpolateVariables(expr string, interval time.Duration, timeRange time.Duration, intervalCalculator intervalv2.Calculator, timeInterval string) string {
+func interpolateVariables(model *QueryModel, interval time.Duration, timeRange time.Duration, intervalCalculator intervalv2.Calculator, timeInterval string) string {
+	expr := model.Expr
 	rangeMs := timeRange.Milliseconds()
 	rangeSRounded := int64(math.Round(float64(rangeMs) / 1000.0))
 
+	var rateInterval time.Duration
+	if model.Interval == varRateInterval || model.Interval == varRateIntervalAlt {
+		rateInterval = interval
+	} else {
+		rateInterval = calculateRateInterval(interval, timeInterval, intervalCalculator)
+	}
+
 	expr = strings.ReplaceAll(expr, varIntervalMs, strconv.FormatInt(int64(interval/time.Millisecond), 10))
 	expr = strings.ReplaceAll(expr, varInterval, intervalv2.FormatDuration(interval))
 	expr = strings.ReplaceAll(expr, varRangeMs, strconv.FormatInt(rangeMs, 10))
```

Comment:
i see that two uses of `intervalv2.FormatDuration` remain in this file. are those ok?